Backend/src/v1/dataset.py

```diff
@@ -45,5 +46,13 @@ def delete_collection(collection: str) -> None:
     OBS, this is permanent!
     """
 
-    # db_delete_collection(collection=collection)
-    raise HTTPException(status_code=501, detail="Not implemented")
+    collections = db_get_all_collections()
+    if collections == []:
+        raise HTTPException(status_code=400, detail="No collections found")
+    
+    for one_collection in collections:
+        if one_collection == collection:
+            db_delete_collection(collection=collection)
+            return
+    
+    raise HTTPException(status_code=400, detail="No collection found")
```

Comment:
I think that it would be good to change from collection to dataset. I used the word collection before but now that everything is called dataset instead I think it is clearer for users if it is called dataset. (Both the HTTPException details and in the docstring)

Reply:
Ok, used collection since that is what the other route in the same file uses.